internal/db/schema/migrations/oss/postgres/29/01_kms_domain_types.up.sql

```diff
@@ -0,0 +1,99 @@
+begin;
+
+-- make the required schema changes to adopt
+-- github.com/hashicorp/go-kms-wrapping/extras/kms/v2 
+-- this migration is from:
+-- https://github.com/hashicorp/go-kms-wrapping/blob/main/extras/kms/migrations/postgres/01_domain_types.up.sql 
+
+create domain kms_private_id as text
+not null
+check(
+  length(trim(value)) > 0
+);
+
+create domain kms_scope_id as text
+check(
+  length(trim(value)) > 0
+);
+
+create domain kms_timestamp as
+  timestamp with time zone
+  default current_timestamp;
+comment on domain kms_timestamp is
+'Standard timestamp for all create_time and update_time columns';
+
+create domain kms_version as bigint
+  default 1
+  not null
+  check(
+   value > 0
+  );
+comment on domain kms_version is
+'standard column for row version';
+
+-- kms_immutable_columns() will make the column names immutable which are passed as
+-- parameters when the trigger is created. It raises error code 23601 which is a
+-- class 23 integrity constraint violation: immutable column  
+create function kms_immutable_columns()
+  returns trigger
+as $$
+declare 
+	col_name text; 
+	new_value text;
+	old_value text;
+begin
+  foreach col_name in array tg_argv loop
+    execute format('SELECT $1.%I', col_name) into new_value using new;
+    execute format('SELECT $1.%I', col_name) into old_value using old;
+  	if new_value is distinct from old_value then
+      raise exception 'immutable column: %.%', tg_table_name, col_name using
+        errcode = '23601', 
+        schema = tg_table_schema,
+        table = tg_table_name,
+        column = col_name;
+  	end if;
+  end loop;
+  return new;
+end;
+$$ language plpgsql;
+
+comment on function
+  kms_immutable_columns()
+is
+  'function used in before update triggers to make columns immutable';
+
+
+create function kms_default_create_time()
+  returns trigger
+as $$
+begin
+  if new.create_time is distinct from now() then
+    new.create_time = now();
+  end if;
+  return new;
+end;
+$$ language plpgsql;
+comment on function
+  kms_default_create_time()
+is
+  'function used to properly set create_time columns';
+
+create or replace function
```

Comment:
```suggestion
create function
```
Would this have caught the naming error for this function?

Reply:
your suggestion would require dropping the function first since it's already associated with the root key tables.... at this point I'm happy with this compromise of "or replace".

Reply:
changing to create